sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -1100,6 +1102,12 @@ inline std::string GetConnectionKey(std::string const& host, CurlTransportOption
 }
 } // namespace
 
+bool CurlConnection::IsValid() const
+{
+  char buffer[32];
```

Comment:
Do we need to read 32 bytes to perform this check?  Could we only peek 1 byte and achieve the same check?

Reply:
1 byte would be enough, yes.